File: lab2/frontend/libs/shared/src/lib/shared/models/product.ts
import { Coordinates } from "./coordinates";
import { Owner } from "./owner";
import { UnitOfMeasureType } from "./unit-of-measure.enum";

export interface Product {
    price: number;
    name: string;
    coordinates: Coordinates;
    manufactureCost: number;
    partNumber: string;
    id: number;
    owner: Owner;
    creationDate: string;
    unitOfMeasure: UnitOfMeasureType;
}

File: lab2/backend/soa-products/src/main/kotlin/com/soa/products/domain/Product.kt
package com.soa.products.domain

import java.time.LocalDate

data class Product(
    val id: Long,
    val name: String,
    val coordinates: Coordinates,
    val creationDate: LocalDate,
    val price: Long,
    val partNumber: String?,
    val manufacturerCost: Long,
    val unitOfMeasure: UnitOfMeasure,
    val owner: Person?,
)

data class Coordinates(
    val x: Int,
    val y: Int,
)

enum class UnitOfMeasure {
    KILOGRAMS,
    METERS,
    MILLILITERS,
    MILLIGRAMS,
}
